pre_commit_dbt/check_model_has_tests_by_group.py

```diff
@@ -0,0 +1,84 @@
+import argparse
+from itertools import groupby
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+
+from pre_commit_dbt.utils import add_filenames_args
+from pre_commit_dbt.utils import add_manifest_args
+from pre_commit_dbt.utils import get_filenames
+from pre_commit_dbt.utils import get_json
+from pre_commit_dbt.utils import get_models
+from pre_commit_dbt.utils import get_tests
+from pre_commit_dbt.utils import JsonOpenError
+from pre_commit_dbt.utils import ParseDict
+
+
+def check_test_cnt(
+    paths: Sequence[str], manifest: Dict[str, Any], test_group: Dict[str, int], test_cnt: int
+) -> int:
+    status_code = 0
+    sqls = get_filenames(paths, [".sql"])
+    filenames = set(sqls.keys())
+
+    # get manifest nodes that pre-commit found as changed
+    models = get_models(manifest, filenames)
+
+    for model in models:
+        tests = list(get_tests(manifest=manifest, obj=model))
+
+        grouped = groupby(
+            sorted(tests, key=lambda x: x.test_name), lambda x: x.test_name
+        )
+        test_dict = {key: list(value) for key, value in grouped}
+        required_test_count = 0
+        for test in test_group:
+            if test_dict.get(test):
+              required_test_count += 1
+        if required_test_count < test_cnt:
+            print(
+                f"{model.model_name}: "
+                f"has only {required_test_count} test(s) from {test_group}.",
+            )
+            status_code = 1
+    return status_code
+
+
+def main(argv: Optional[Sequence[str]] = None) -> int:
+    parser = argparse.ArgumentParser()
+    add_filenames_args(parser)
+    add_manifest_args(parser)
+    parser.add_argument(
+        "--tests",
+        nargs="+",
+        required=True,
+        help="List of acceptable tests.",
+    )
+    parser.add_argument(
+        "--test-cnt",
+        type=int,
+        default=1,
+        help="Minimum number of tests required.",
+    )
+
+    args = parser.parse_args(argv)
+
+    try:
+        manifest = get_json(args.manifest)
+    except JsonOpenError as e:
+        print(f"Unable to load manifest file ({e})")
+        return 1
+
+    try:
```

Comment:
This is not needed since argparse take care of type. Test_cnt variable is not even used (see line 79), so you can delete this whole try: part

Reply:
And this is also causing that tests fails since this exception is not tested anywhere -> test coverage < 100%